### PR TITLE
Generalize ff interaction

### PR DESCRIPTION
**Pull request by ehhartmann**
Deal with problems discussed in #359 and facilitate future addition of non-amber force-fields to kimmdy.

- [x] expose reference to residuetypes file in config
- [x] add optional radicals section to config
- [x] add tests

### CODE REVIEW

**Comment by ehhartmann** [comment]
2 failed, 94 passed, ready for review

**Comment by jmbuhr** [comment]
Which are the failing tests? tox looks fine. I'd advise to run black before the review so that we have common formatting, otherwise we might get confused with line numbers during it.

**Comment by ehhartmann** [comment]
Integration as always. It's formatted with the latest black version now. To me the docs action failure looks like some parts of the action are outdated.

**Comment by ehhartmann** [comment]
To use the new features of this PR, the plugins (at least both HAT ones) would have to be updated as well

**Comment by jmbuhr** [comment]
ah, looks like the doc parser is now attempting to parse every heading in the docstring as a proper semantic section, but https://github.com/hits-mbm-dev/kimmdy/blob/ce1daf0e68627c04e16c878e0d4d8f897ad5c833/src/kimmdy/parsing.py#L108-L109 is just a random heading I threw in, so this just has to not be a section by removing the `---`.

**Comment by jmbuhr** [comment]
would you like me to do small fixes directly in the PR when I review or just comment instead?

**Comment by ehhartmann** [comment]
Just change it

File: src/kimmdy/topology/topology.py [hunk 1]
```diff
@@ -86,7 +89,25 @@ def __init__(self, header: tuple[str, str], atomics: dict) -> None:
...
+                atom = self.atoms.get(radical)
+                if atom:
+                    self.radicals[radical] = atom
+                else:
+                    logger.debug(
+                        f"Atom nr {radical} from 'radicals' section in config file not in topology. Ignoring this entry."
+                    )
+        else:
+            self.find_radicals()
+        self.set_atom_is_radical()
```
Comment (jmbuhr) [inline src/kimmdy/topology/topology.py:110]: Ah, I see now how these functions have to be used independently of each other, so it's just about finding better names, not merging them.

File: src/kimmdy/topology/topology.py [hunk 2]
```diff
@@ -600,14 +631,14 @@ def __init__(
...
-        self.ff = FF(top)
+        self.ff = FF(top, residuetypes_path)
         self._parse_molecules()
         self.parametrizer = parametrizer
         self._check_is_reactive_molecule = is_reactive_predicate_f
 
         self.needs_parameterization = False
 
-        self._merge_moleculetypes()
+        self._merge_moleculetypes(radicals)
```
Comment (jmbuhr) [inline src/kimmdy/topology/topology.py:641]: I think I'll have some minor cleanups about what part is responsible for keeping track of the radical list such that the moleculetype doesn't need to check again if the radical it gets are for its own use.

**Review by jmbuhr — COMMENTED** [review]
looking good, just some minor questions I can then also fix myself if you approve.

File: src/kimmdy/coordinates.py [hunk 3]
```diff
@@ -475,7 +475,8 @@ def merge_top_moleculetypes_slow_growth(
                 molB.atoms[k].typeB = "H1"
 
     # update is_radical attribute of Atom objects in topology
-    molB.test_for_radicals()
+    molB.find_radicals()
```
[the remark this answers is not in the document]
Comment (jmbuhr) [reply src/kimmdy/coordinates.py:478]: nvm., invalidated by my later comment

**Comment by ehhartmann** [comment]
For radicals, the workflow is different between writing `radicals: ''` and not having a radicals section at all (#59) because in the missing section case, we use the bondorder to guess radicals. For the modify-top tool I thought it was a reasonable default to say there are no radicals but I wouldn't do that for a kimmdy run. If you want to have it consistent, I would have no default in both.

Regarding residuetypes, it is similar. No section means we want to use the one in the force field. Having aminoacids.rtp as default would mean we need it in the rundir. 

Feel free to rename the functions for finding and setting the atoms attribute. If you find ways to make the same logic more elegant I would be happy. It felt quite awkward to do it like this.

File: src/kimmdy/topology/topology.py (hunk 1 above)
Comment (jmbuhr) [reply src/kimmdy/topology/topology.py:110]: the `set_atom_is_radical` method was not necessary in the end. Refactored into the `find_radicals` method and the `Moleculetype.__init__`

**Review by jmbuhr — APPROVED** [review]

**Pull request merged**